### PR TITLE
feature: Import pattern configurations CY-3356

### PR DESCRIPTION
**Pull request by pauloribeiro-codacy**
Documentation for the feature that allows importing the pattern configurations from a different repository.

See [CY-3162](https://codacy.atlassian.net/browse/CY-3162) and [CY-3356](https://codacy.atlassian.net/browse/CY-3356) for more details.

### CODE REVIEW

**Comment by pauloribeiro-codacy** [comment]
Here is a preview of the relevant section in the page about configuring code patterns:

https://github.com/codacy/docs/blob/feature/import-patterns-CY-3356/docs/repositories-configure/code-patterns.md#importing-pattern-configurations-from-another-repository--idimport-patterns

**Comment by machadoit** [comment]
I like it! LGTM! 👍

**Review by machadoit — APPROVED** [review]

**Comment by pauloribeiro-codacy** [comment]
> I like it! LGTM!

Thanks for reviewing @machadoit! I made a few changes, namely adding labels in bold at the start of each sentence describing how the import works to improve readability and moving the list inside an "Important" card. What do you think?

**Review by machadoit — DISMISSED** [review]

**Review by heliocodacy — DISMISSED** [review]

File: docs/repositories-configure/code-patterns.md [hunk 1]
```diff
@@ -64,7 +64,6 @@ Importing tool and pattern configurations from another repository can help you b
 !!! important
     Consider the following when importing tool and pattern configurations:
 
-    -   **Permissions:** You must have admin permissions on the source repository.
```
Comment (pauloribeiro-codacy) [inline docs/repositories-configure/code-patterns.md]: @machadoit, please review the rationale for completely removing this line in this commit message: https://github.com/codacy/docs/pull/397/commits/857f9d780a455c9e6e7c139b94801c95e37b7319

**Review by machadoit — DISMISSED** [review]

**Review by inesgomas — APPROVED** [review]

**Comment by pauloribeiro-codacy** [comment]
I had the idea of trying to use a simple diagram to illustrate how the import patterns feature works. Let me know what you think, do you think the table with the diagram is useful? Are you able to make sense of it? :slightly_smiling_face: 

image

**Review by lolgab — CHANGES_REQUESTED** [review]
@pauloribeiro-codacy Did you notice that Github can't render the table and the list you used?
In case you did, is it a problem of github only?
I noticed that other tables are in the code as html tables while this is using the markdown specific syntax. Is this intended?

**Comment by pauloribeiro-codacy** [comment]
> @pauloribeiro-codacy Did you notice that Github can't render the table and the list you used?
> In case you did, is it a problem of github only?
> I noticed that other tables are in the code as html tables while this is using the markdown specific syntax. Is this intended?
> 
> **EDIT**
> apparently I didn't see your screenshot above  . Still, why not a html table as in the other tables?

Hello @lolgab,

Yes, this is because GitHub doesn't render the [Admonition extension](https://python-markdown.github.io/extensions/admonition/) syntax. If you notice, it's not just the table that is rendered incorrectly but the list and paragraphs as well.

Regarding the choice of table syntax, I normally start with the Markdown syntax and only switch to HTML if the table becomes more complex. For example, if it becomes too big, if I need to merge cells, or if I need to place more complex structures inside the cells. For simple, more straightforward tables I find that Markdown is just easier to write and change. :sweat_smile:

**Review by lolgab — DISMISSED** [review]

**Pull request merged**